planner/core/exhaust_physical_plans.go

```diff
@@ -547,14 +547,28 @@ func (p *LogicalJoin) buildRangeForIndexJoin(indexInfo *model.IndexInfo, innerPl
 		return nil, nil, nil
 	}
 
-	// We should guarantee that all the join's equal condition is used.
-	for _, eqCond := range eqConds {
+	// Guarantee res.AccessConds is not empty.
+	if len(res.AccessConds) == 0 {
+		return nil, nil, nil
+	}
+
+	// Find invalid fake condition and modify the joinKey's idxOff to -1.
+	var invalidFakeConds []expression.Expression
+	for i, eqCond := range eqConds {
 		if !expression.Contains(res.AccessConds, eqCond) {
-			return nil, nil, nil
+			keyOff2IdxOff[i] = -1
+			invalidFakeConds = append(invalidFakeConds, eqCond)
+		}
+	}
+
+	// Filter out invalidFakeConds from res.RemainedConds.
+	for _, cond := range res.RemainedConds {
+		if !expression.Contains(invalidFakeConds, cond) {
```

Comment:
Why should we add this check? Why we cannot just append `res.RemainedConds` into `remained`? could you please give an example for illustration?

Reply:
Oh, I see, it contains fake equal conditions, so it should be removed.